lib/games/publishToTopic.js

```diff
@@ -0,0 +1,52 @@
+const http = require('../util/http.js').func
+
+exports.required = ['universeId', 'topic', 'data']
+exports.optional = ['jar']
+
+// Docs
+/**
+ * ☁️ Publish a message to a subscribed topic.
+ * @category Game
+ * @alias postToTopic
+ * @param {number} universeId - The id of the universe.
+ * @param {string} topic - The name of the topic.
+ * @param {Object | string} data - The data to post.
+ * @returns {Promise<boolean>}
+ * @example const noblox = require("noblox.js")
+ * const data = { targetUser: "1210019099" }
+ * await noblox.publishToTopic(2152417643, "MyTopic", data)
+**/
+
+function publishToTopic (universeId, topic, data, jar) {
+  return new Promise((resolve, reject) => {
+
+    const httpOpt = {
+      url: `//apis.roblox.com/messaging-service/v1/universes/${universeId}/topics/${topic}`,
+      options: {
+        json: true,
+        resolveWithFullResponse: true,
+        jar,
+        method: 'POST',
+        body: { 'message': JSON.stringify(data) },
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      }
+    }
+
+    return http(httpOpt)
+      .then(function ({ statusCode }) {
+        switch (statusCode) {
+          case 200: resolve(true)
+          case 400: reject(new Error(`[${statusCode}] Bad Request, check what data is being sent`))
+          case 401: reject(new Error(`[${statusCode}] Unauthorized`))
+          case 403: reject(new Error(`[${statusCode}] Forbidden, Possible insufficient API key access permissions`))
+          case 500: reject(new Error(`[${statusCode}] Internal server error`))
+        }
+      }).catch(reject)
+  })
```

Comment:
Error handling should probably be handled like in this file: [lib/friends/acceptFriendRequest.js](https://github.com/noblox/noblox.js/blob/master/lib/friends/acceptFriendRequest.js#L37-L49). If I'm wrong, you don't actually need the format string and can just manually enter the error codes which might be slightly more performant.

Reply:
I made a later commit to replace the switch case with just an if else.
I'm not sure what you mean that I don't need to format the string?

Reply:
I just mean errors should probably be reported as they come from Roblox, and not what we currently understand those errors to be. The file I linked has dynamic error handling that should pull the errors from the API instead of being hard coded in.

Reply:
Alright, I completely agree. I'll make a commit for the errors to be handled like they are in acceptFriendRequest.

Reply:
Error handling is now updated to give proper errors. The errors this API returns are very different than others so I had to spent a little more time figuring them out.

Valid API key that has any sort of invalid access permissions/scopes will return:
Error: [403] InsufficientScope Scope not authorized.

Invalid API key will return:
Error: [401] Unauthorized Invalid API Key